language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/AbstractCodeActionProvider.java

```diff
@@ -116,10 +115,29 @@ public final boolean isDiagBasedSupported() {
      * @param uri          uri
      * @return {@link CodeAction}
      */
-    public static CodeAction createQuickFixCodeAction(String commandTitle, List<TextEdit> edits, String uri) {
+    public static CodeAction createCodeAction(String commandTitle, List<TextEdit> edits, String uri) {
         List<Diagnostic> diagnostics = new ArrayList<>();
         CodeAction action = new CodeAction(commandTitle);
-        action.setKind(CodeActionKind.QuickFix);
+        action.setEdit(new WorkspaceEdit(Collections.singletonList(Either.forLeft(
+                new TextDocumentEdit(new VersionedTextDocumentIdentifier(uri, null), edits)))));
+        action.setDiagnostics(CodeActionUtil.toDiagnostics(diagnostics));
+        return action;
+    }
+
+    /**
+     * Returns a Code action.
+     * 
+     * @param commandTitle      title of the code action
+     * @param edits             edits to be added in the code action
+     * @param uri               uri
+     * @param codeActionKind    kind of the code action
+     * @return
+     */
+    public static CodeAction createCodeAction(String commandTitle, List<TextEdit> edits, String uri,
+                                              String codeActionKind) {
```

Comment:
@nadeeshaan 
The `CodeActionKind`'s are strings. Are you suggesting to use something like, changing the parameter to `CodeActionKind codeActionKind` and  access it as `codeActionKind.getString()` since we don not ave other methods in `CodeActionKind.java`?

Reply:
Sorry, my bad, I was mistakenly considered the `CodeActionKind` being an enum which is not. Your approach is correct. +1 to proceed